tensorflow/compiler/mlir/lite/transforms/dilated_conv.h

```diff
@@ -297,12 +318,31 @@ ConvertTFDilatedConvOp<Conv2dOpTy>::ExtractDilationsAttrFromBlockShape(
     if (stb_bs_attr.getValue({i}) != bts_bs_attr.getValue({i})) return {};
   }
 
+  int dilation_h_factor, dilation_w_factor;
   // Set dilation factor.
-  if (stb_bs_attr.getNumElements() < 2) return {};
-  int dilation_h_factor =
-      stb_bs_attr.getValue({0}).cast<IntegerAttr>().getInt();
-  int dilation_w_factor =
-      stb_bs_attr.getValue({1}).cast<IntegerAttr>().getInt();
+  if (stb_bs_attr.getNumElements() >= 2) {
+    dilation_h_factor = stb_bs_attr.getValue({0}).cast<IntegerAttr>().getInt();
+    dilation_w_factor = stb_bs_attr.getValue({1}).cast<IntegerAttr>().getInt();
+  } else if (stb_bs_attr.getNumElements() == 1) {
+    // For 1d conv, `tf.nn.convolution` expands NWC to NHWC format after
+    // `SpaceToBatchND`. Therefore, `block_shape` of `stb_op` only has one
+    // dilation factor of W dim, and dilation factor of H dim is set to 1.
+    if (expand_axis == 1) {
+      // NWC -> NHWC
+      dilation_h_factor = 1;
+      dilation_w_factor =
+          stb_bs_attr.getValue({0}).cast<IntegerAttr>().getInt();
+    } else if (expand_axis == 2) {
+      // NHC -> NHWC
+      dilation_h_factor =
+          stb_bs_attr.getValue({0}).cast<IntegerAttr>().getInt();
+      dilation_w_factor = 1;
+    } else {
```

Comment:
nit, you can reduce those two returns with only one at the end.

Reply:
Will do!

Reply:
Updated. Please see if this is better.

Reply:
Yeap! It looks better.